File: JavaScript/Tecnicatura3JS/TSleccion01/Leccion09/Herencia/Persona.js
class Persona {

    static contador = 0;

    constructor(nombre,apellido,edad) {
        this._idPersona = ++Persona.contador;
        this._nombre = nombre
        this._apellido = apellido
        this._edad = edad
    }

    get idPersona() {
        return this._idPersona;
    }

    get nombre(){
        return this._nombre
    }
    get apellido(){
        return this._apellido
    }
    get edad(){
        return this._edad
    }

    set nombre(nombre){
        this._nombre = nombre
    }
    set apellido(apellido){
        this._apellido = apellido
    }
    set edad(edad){
        this._edad = edad
    }

    toString(){
        return this.idPersona() + " " + this._nombre() + " " + this._apellido() + " " + this._edad();
    }
    
}

let p1 = new Persona("ale","ale",24);

console.log(p1.toString())

File: TECNICATURA/JAVA/Clase08/Persona.java

package domain;


public class Persona {

    //Cargamos los atributos
    private int idPersona;
    private static int contadorPersona;
    private String nombre;
    
    //Constructor
    public Persona(String nombre){
        this.nombre = nombre;
        //Incrementar el contador por cada objeto nuevo
        Persona.contadorPersona++; //no utilizar el operador this
        //vamos a asignar un nuevo valor a la variable idPersona
        this.idPersona = Persona.contadorPersona;
    }
     public static int getContadorPersona() {
        return contadorPersona;
    }

    public static void setContadorPersona(int aContadorPersona) {
        contadorPersona = aContadorPersona;
    }
    
    public int getIdPersona() {
        return this.idPersona;
    }

    public void setIdPersona(int idPersona) {
        this.idPersona = idPersona;
    }

    public String getNombre() {
        return this.nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    @Override //esta notación agrega info extra al método que estamos definiendo
    public String toString() {
        return "Persona{" + "idPersona=" + idPersona + ", nombre=" + nombre + '}';
    }
       
}
